_posts/2022-10-18-torch-linalg/torch-linalg.Rmd

```diff
@@ -0,0 +1,632 @@
+
+
+## Regression for weather prediction
+
+The dataset we'll use is available from the [UCI Machine Learning Repository](http://archive.ics.uci.edu/ml/machine-learning-dAtAbases/00514/Bias_correction_ucl.csv).
+
+```{r}
+library(torch)
+library(dplyr)
+library(readr)
+library(zeallot)
+
+uci <- "https://archive.ics.uci.edu"
+ds_path <- "ml/machine-learning-databases/00514"
+ds_file <- "Bias_correction_ucl.csv"
+
+download.file(
+ file.path(uci, ds_path, ds_file),
+ destfile = "resources/matrix-weather.csv"
+)
+
+weather_df <- read_csv("resources/matrix-weather.csv") %>%
+  na.omit()
+weather_df %>% glimpse()
+```
+
+    Rows: 7,588
+    Columns: 25
+    $ station           <dbl> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,…
+    $ Date              <date> 2013-06-30, 2013-06-30,…
+    $ Present_Tmax      <dbl> 28.7, 31.9, 31.6, 32.0, 31.4, 31.9,…
+    $ Present_Tmin      <dbl> 21.4, 21.6, 23.3, 23.4, 21.9, 23.5,…
+    $ LDAPS_RHmin       <dbl> 58.25569, 52.26340, 48.69048,…
+    $ LDAPS_RHmax       <dbl> 91.11636, 90.60472, 83.97359,…
+    $ LDAPS_Tmax_lapse  <dbl> 28.07410, 29.85069, 30.09129,…
+    $ LDAPS_Tmin_lapse  <dbl> 23.00694, 24.03501, 24.56563,…
+    $ LDAPS_WS          <dbl> 6.818887, 5.691890, 6.138224,…
+    $ LDAPS_LH          <dbl> 69.45181, 51.93745, 20.57305,…
+    $ LDAPS_CC1         <dbl> 0.2339475, 0.2255082, 0.2093437,…
+    $ LDAPS_CC2         <dbl> 0.2038957, 0.2517714, 0.2574694,…
+    $ LDAPS_CC3         <dbl> 0.1616969, 0.1594441, 0.2040915,…
+    $ LDAPS_CC4         <dbl> 0.1309282, 0.1277273, 0.1421253,…
+    $ LDAPS_PPT1        <dbl> 0.0000000, 0.0000000, 0.0000000,…
+    $ LDAPS_PPT2        <dbl> 0.000000, 0.000000, 0.000000,…
+    $ LDAPS_PPT3        <dbl> 0.0000000, 0.0000000, 0.0000000,…
+    $ LDAPS_PPT4        <dbl> 0.0000000, 0.0000000, 0.0000000,…
+    $ lat               <dbl> 37.6046, 37.6046, 37.5776, 37.6450,…
+    $ lon               <dbl> 126.991, 127.032, 127.058, 127.022,…
+    $ DEM               <dbl> 212.3350, 44.7624, 33.3068, 45.7160,…
+    $ Slope             <dbl> 2.7850, 0.5141, 0.2661, 2.5348,…
+    $ `Solar radiation` <dbl> 5992.896, 5869.312, 5863.556,…
+    $ Next_Tmax         <dbl> 29.1, 30.5, 31.1, 31.7, 31.2, 31.5,…
+    $ Next_Tmin         <dbl> 21.2, 22.5, 23.9, 24.3, 22.5, 24.0,…
+
+The way we're framing the task, nearly everything in the dataset serves as a predictor. As target, we'll use `Next_Tmax`, the maximal temperature reached on the subsequent day. This means we need to remove `Next_Tmin` from the set of predictors, as it would make for too powerful of a clue. We'll do the same for `station`, the weather station id, and `Date`. This leaves us with twenty-one predictors, including measurements of actual temperature (`Present_Tmax`, `Present_Tmin`), model forecasts of various variables (`LDAPS_*`), and auxiliary information (`lat`, `lon`, and `` `Solar radiation` ``, among others).
```

Comment:
"As target" feels like it's missing an article (e.g. As "a" target, as "the" target, as "our" target). If this is a common way of stating it in technical documentation, though, of course ignore my ignorance.

Reply:
OMG, I'm sure I'm much more ignorant than you, in that respect (you're not at all, I'm sure!).
Will change to "a"